src/main/java/org/jabref/logic/importer/fetcher/ISIDOREFetcher.java

```diff
@@ -0,0 +1,223 @@
+                }
+
+                return Collections.singletonList(xmlItemToBibEntry(document));
+            } catch (Exception e) {
+                // Handle parsing exceptions
+                try {
+                    throw new FetcherException("Issue with parsing link");
+                } catch (FetcherException ex) {
+                    throw new RuntimeException(ex);
+                }
+            }
+        };
+    }
+
+    @Override
+    public URL getUrlForIdentifier(String identifier) throws URISyntaxException, MalformedURLException, FetcherException {
+        identifier = identifier.trim();
+        // this allows the user to input only the six-digit code at the end.
+        if (identifier.length() == 6) {
+            identifier = "https://isidore.science/document/10670/1." + identifier;
+        } else if (identifier.length() == 8) {
+            // allows the user to put in the eight digits including the "1."
+            identifier = "https://isidore.science/document/10670/" + identifier;
+        }
+        // Throw an error if this is not the starting link
+        if (identifier.startsWith("https://isidore.science/document/10670/1.") && (identifier.length() == LINKLENGTH)) {
+            this.URL = identifier;
+            // change the link to be the correct link for the api.
+            identifier = identifier.replace("/document/", "/resource/content?uri=");
+            identifier = identifier.replace("https://isidore.science/", "https://api.isidore.science/");
+            return new URL(identifier);
+        } else {
+            throw new FetcherException("Could not construct url for ISIDORE");
+        }
+    }
+
+    @Override
+    public Parser getParser() {
+        return this.parser;
+    }
+
+    private BibEntry xmlItemToBibEntry(Document document) {
+        Element itemElement = document.getDocumentElement();
+        return new BibEntry(getType(itemElement.getElementsByTagName("types").item(0).getChildNodes()))
+                .withField(StandardField.TITLE, itemElement.getElementsByTagName("title").item(0).getTextContent())
+                .withField(StandardField.AUTHOR, getAuthor(itemElement.getElementsByTagName("enrichedCreators").item(0)))
+                .withField(StandardField.YEAR, itemElement.getElementsByTagName("date").item(0).getChildNodes().item(1).getTextContent().substring(0, 4))
+                .withField(StandardField.JOURNAL, getJournal(itemElement.getElementsByTagName("dc:source")))
+                .withField(StandardField.PUBLISHER, getPublishers(itemElement.getElementsByTagName("publishers").item(0)))
+                .withField(StandardField.DOI, getDOI(itemElement.getElementsByTagName("ore").item(0).getChildNodes()))
+                .withField(StandardField.URL, this.URL);
+    }
+
+    private String getDOI(NodeList list) {
+        for (int i = 0; i < list.getLength(); i++) {
+            if (list.item(i).getTextContent().contains("DOI:")) {
```

Comment:
Extract `list.item(i).getTextContent()` as variable `content`. (You can use IntelliJ <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>V</kbd> for that)